File: app/src/main/java/br/com/ufg/www/events/domain/BaseActivity.java
package br.com.ufg.www.events.domain;

import android.app.Activity;
import android.support.annotation.NonNull;
import android.support.design.widget.Snackbar;

public abstract class BaseActivity extends Activity implements UIFeedback {

    @Override
    public void showMessage(@NonNull String message) {
        Snackbar.make(findViewById(android.R.id.content), message, Snackbar.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/br/com/ufg/www/events/domain/BaseActivity.kt
package br.com.ufg.www.events.domain

import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.snackbar.Snackbar

abstract class BaseActivity : AppCompatActivity(), UIFeedback {

    override fun showMessage(message: String) = Snackbar.make(findViewById<View>(android.R.id.content), message, Snackbar.LENGTH_SHORT).show()

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (android.R.id.home == item?.itemId) {
            onBackPressed()
        }
        return super.onOptionsItemSelected(item)
    }

}